lib/prometheus/middleware/collector.rb

```diff
@@ -91,6 +91,7 @@ def strip_ids_from_path(path)
         path
           .gsub(%r{/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}(/|$)}, '/:uuid\\1')
           .gsub(%r{/\d+(/|$)}, '/:id\\1')
+          .gsub(%r{/[0-9a-zA-Z=]+--[0-9a-f]+(/|$)}, '/:signed_id\\1')
```

Comment:
Thank you for your PR!

I'm a little worried that this Regex is a bit broad. It'll match almost any part of a path that has `--` in the middle. This is admittedly rare, but i'm wondering if we can narrow it down a bit more, probably by specifying the length of these parts, like we do for `uuid`.

I have no experience with ActiveStorage unfortuantely, but reading through the code, i believe this is where the Signed ID gets generated: https://github.com/rails/rails/blob/afc79e3cb0906d06b035ef30411c44ddc2132409/activesupport/lib/active_support/message_verifier.rb#L186-L203

And if i'm reading that correctly, i'd expect the first half of the signed ID (before the `--`) can be any length, but the digest should always be the same length (40 chars), right?

I notice your test case has a shorter digest, i'm not sure whether you're using an actual Signed ID generated by Rails, which would invalidate my suggestion, or if it was manual random typing...

Thoughts?

Reply:
Thank you for reviewing the PR,
I completely agree with that point, we should make the regex stricter.
Regarding the signature id, though my example was bogus, unfortunately lengths is not always 40, thought the default is SHA1, but it could be changed:
https://github.com/rails/rails/blob/afc79e3cb0906d06b035ef30411c44ddc2132409/activesupport/lib/active_support/message_verifier.rb#L109

Most likely to something like SHA-256 which is 64 chars but extremely rare (rarer than "having -- in path" rare) for anyone to change it to anything less than MD5 which is 32.

Also the first part (before --) also should be a valid base64 encoded string so a not a complex overkill  regex might look like this:
`(?=(.{4})*--)[A-Za-z0-9]+={0,2}--[0-9a-f]{32,}`

I can also add negative test case to the specs.